StringCalculator/Services/NumberService.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace StringCalculator.Services
+{
+    public class NumberService : INumberService
+    {
+        private List<string> _delimiters = new List<string>() { ",", "\n" };
+        private string[] _delimiterSurroundingStrings = { "[", "][", "]" };
+        private string[] _stringNumbers;
+
+        private const string DelimiterAndNumberSeperator = "\n";
+        private const string DelimiterStartString = "//";
+        private const int MaximumNumber = 1000;
+        private const int DelimiterIndicatorIndex = 2;
+
+        public List<int> ParseNumbers(string inputString)
+        {
+            ExtractDelimitersAndNumbers(inputString);
+
+            var parsedNumbers = ValidateAndParseNumbers();
+
+            return parsedNumbers;
+        }
+
+        private void ExtractDelimitersAndNumbers(string inputString)
+        {
+            var seperatorIndex = GetSeperatorIndex(inputString);
+
+            if (inputString.StartsWith(DelimiterStartString))
+            {
+                inputString = inputString[DelimiterIndicatorIndex..];
+
+                var allDelimiters = inputString[..(seperatorIndex - 1)].Split(_delimiterSurroundingStrings, StringSplitOptions.RemoveEmptyEntries);
+
+                _delimiters.AddRange(allDelimiters);
+
+                _stringNumbers = inputString[seperatorIndex..].Split(_delimiters.ToArray(), StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Not a fan of using the instance variables here - rather be returning the list of string numbers, as it's easier to follow (and thus more readable).